### PR TITLE
Onbarding openshift-eng/ofcir repo

### PR DESCRIPTION
**Pull request by tdomnesc**
CI config for the new OFCIR project

### CODE REVIEW

**Comment by derekhiggins** [comment]
/lgtm

> /pj-rehearse
hmm, I wonder why this didn't run a rehearse of "pull-ci-openshift-eng-ofcir-main-unit"

**Comment by derekhiggins** [comment]
@tdomnesc I see that the unit tests are failing is this something we need to fix in the Makefile or on this PR?

**Comment by tdomnesc** [comment]
> @tdomnesc I see that the unit tests are failing is this something we need to fix in the Makefile or on this PR?

It's things particular to the CI environment so I think we need to fix them here. The failings so far: a vendor dir is created somehow, and it tries to use /.cache as a cache dir

File: ci-operator/config/openshift-eng/ofcir/openshift-eng-ofcir-main.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.18
```
Comment (andfasano) [inline ci-operator/config/openshift-eng/ofcir/openshift-eng-ofcir-main.yaml:5]: Can we use `rhel-8-release-golang-1.18-openshift-4.12` as a tag?

File: ci-operator/config/openshift-eng/ofcir/openshift-eng-ofcir-main.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: unit
+  commands: |
+    XDG_CACHE_HOME="/tmp/cache" \
+    GOCACHE="/tmp/cache" \
+    GOFLAGS="-mod=readonly" \
+    make unit-tests
```
Comment (andfasano) [inline ci-operator/config/openshift-eng/ofcir/openshift-eng-ofcir-main.yaml:20]: Not clear why these env vars are needed. We can review their usage later anyway

**Pull request merged**